### PR TITLE
Add photon vision

### PR DESCRIPTION
**Pull request by aidnem**
Adds basic photon vision implementation.

Works in sim, and has been demonstrated in HITL that a camera can output a pose estimate if all pose-rejection is disabled (this is as close to a proof of concept as we can get with only one camera).

Demonstration in sim can be seen in videos posted under #7

### CODE REVIEW

**Comment by aidnem** [comment]
@jkleiber In my opinion, the requirements of #7 were just a barebones system and should therefore be merged in now. Making vision really good can be done later, probably as part of #14

File: src/main/java/frc/robot/BuildConstants.java
Comment (jkleiber) [inline src/main/java/frc/robot/BuildConstants.java:1]: Can we gitignore this file

**Review by jkleiber — APPROVED** [review]
Approved with 1 non blocking comment

File: src/main/java/frc/robot/BuildConstants.java
Comment (aidnem) [reply src/main/java/frc/robot/BuildConstants.java:1]: I'll add it to my cleanup issue #18

**Pull request merged**